### PR TITLE
buildGlobalParameterDefinition implementation & testing

### PR DESCRIPTION
**Pull request by schaable**
Part of https://github.com/NomicFoundation/hardhat/issues/5379

This should be easy to review by going through it commit by commit.

### CODE REVIEW

File: v-next/core/src/config.ts [hunk 1]
```diff
@@ -57,11 +58,11 @@ export function overrideTask(
...
  * Defines a global parameter.
  */
-export function globalParameter(options: {
+export function globalParameter<T extends ParameterType>(options: {
   name: string;
   description: string;
-  parameterType: ParameterType;
-  defaultValue: any;
+  parameterType?: T;
+  defaultValue: ParameterTypeToValueType<T>;
```
Comment (schaable) [inline v-next/core/src/config.ts:65]: @alcuadrado I've added the type validation we use in the task parameters, I think it makes sense to have it here as well.

File: v-next/core/src/internal/global-parameters.ts [hunk 2]
```diff
@@ -48,21 +58,47 @@ export function buildGlobalParameterMap(
...
+  name,
+  description,
+  parameterType,
+  defaultValue,
+}: {
   name: string;
   description: string;
-  parameterType: ParameterType;
-  defaultValue: any;
+  parameterType?: T;
```
Comment (schaable) [inline v-next/core/src/internal/global-parameters.ts:69]: @alcuadrado I think `parameterType` should be just `type` to be consistent with task parameters. wdyt? I'll change it on a follow up PR if you agree.

**Review by alcuadrado — APPROVED** [review]
LGTM. And agreed with both of your comments

File: v-next/core/src/internal/global-parameters.ts (hunk 2 above)
Comment (schaable) [reply v-next/core/src/internal/global-parameters.ts:69]: Added an [issue](https://github.com/NomicFoundation/hardhat/issues/5415) to track it.

**Pull request merged**